### PR TITLE
[CSL-1279] Part II: Get rid of the 'ssc' param everywhere

### PR DESCRIPTION
**Pull request by neongreen**
...and some minor changes. Stay tuned for part III:

* `Ssc` and other (useless now) classes will be removed
* `SscConstraint` will go away
* `SscBlock` will be made a type instead of a newtype
* `SscAlgo` will be removed
* `cardano-sl-ssc` and `cardano-sl-godtossing` will be merged

### CODE REVIEW

**Comment by neongreen** [comment]
P.S. @ksaric I hope you're happy :⁠wink⁠:

File: lib/cardano-sl.cabal [hunk 1]
```diff
@@ -335,6 +335,7 @@ library
                       , cardano-sl-core
                       , cardano-sl-db
                       , cardano-sl-godtossing
+                      , cardano-sl-godtossing
```
Comment (gromakovsky) [inline lib/cardano-sl.cabal]: _mda_

File: lib/src/Pos/Util/JsonLog.hs [hunk 2]
```diff
@@ -127,7 +129,7 @@ $(deriveJSON defaultOptions ''JLTxR)
 $(deriveJSON defaultOptions ''JLMemPool)
 
 -- | Return event of created block.
-jlCreatedBlock :: (BiSsc, HasConfiguration) => Block -> JLEvent
+jlCreatedBlock :: (HasConfiguration, Bi BlockHeader) => Block -> JLEvent
```
Comment (gromakovsky) [inline lib/src/Pos/Util/JsonLog.hs]: I am 99.9% sure `Bi BlockHeader` is redundant.

**Comment by neongreen** [comment]
> All checks have passed

> Conflicting files
> `lib/src/Pos/Launcher/Resource.hs`
> `lib/test/Test/Pos/Block/Logic/Mode.hs`

suka ty

**Pull request merged**